### PR TITLE
Polish container image extensions and doc

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: docs/src/main/asciidoc/container-image.adoc [hunk 1]
```diff
@@ -9,16 +9,16 @@ include::./attributes.adoc[]
 
 Quarkus provides extensions for building (and pushing) container images. Currently it supports:
 
-- jib
-- docker
-- s2i
+- JIB
+- Docker
+- S2I
```
Comment (machi1990) [inline docs/src/main/asciidoc/container-image.adoc]: Should we link to each extension's section?

File: docs/src/main/asciidoc/container-image.adoc [hunk 2]
```diff
@@ -69,69 +72,46 @@ The creation of such objects is being taken care of by the Quarkus kubernetes ex
...
-
 == Pushing
 
-If the execution is set to `push` then after the build, the image will also be pushed to an image registry.
-The registry can be specified using `quarkus.container-image.registry` and will default to `docker.io`.
+To build a container image for your project, `quarkus.container-image.push=true` needs to be set using any of the ways that Quarkus supports.
+
+[source, subs=attributes+]
+----
+mvn clean package -Dquarkus.container-image.push=true
```
Comment (machi1990) [inline docs/src/main/asciidoc/container-image.adoc]:
```suggestion
./mvnw clean package -Dquarkus.container-image.push=true
```

File: extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java [hunk 3]
```diff
@@ -135,6 +135,9 @@ private Containerizer createContainerizer(ContainerImageConfig containerImageCon
         ImageReference imageReference = getImageReference(containerImageConfig, applicationInfo);
 
         if (pushRequested || containerImageConfig.push) {
+            if (!containerImageConfig.registry.isPresent()) {
+                log.info("No container image registry was set, so 'docker.io' will be used");
```
Comment (machi1990) [inline extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java]:
```suggestion
                log.info("No container image registry was set, defaulting to 'docker.io' ");
```

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java [hunk 4]
```diff
@@ -118,6 +118,9 @@ private void createContainerImage(ContainerImageConfig containerImageConfig, Doc
         log.infof("Pushed container image %s (%s)\n", image, reader.getImageId());
 
         if (pushRequested || containerImageConfig.push) {
+            if (!containerImageConfig.registry.isPresent()) {
+                log.info("No container image registry was set, so 'docker.io' will be used");
```
Comment (machi1990) [inline extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java]:
```suggestion
                log.info("No container image registry was set, defaulting to 'docker.io'");
```

File: docs/src/main/asciidoc/container-image.adoc (hunk 2 above)
Comment (machi1990) [reply docs/src/main/asciidoc/container-image.adoc]: We can use the wrapper in the Pushing section too, https://github.com/quarkusio/quarkus/blob/f16f9c9e208ab7d127d8851e8ac54633cc7df857/docs/src/main/asciidoc/container-image.adoc#pushing

File: docs/src/main/asciidoc/container-image.adoc [hunk 5]
```diff
@@ -9,16 +9,16 @@ include::./attributes.adoc[]
...
+- S2I
 
 == Container Image extensions
 
 === JIB
 
-The extension `quarkus-container-image-jib` is powered by https://github.com/GoogleContainerTools/jib[Jib] for performing container builds.
-The major benefit of using Jib with Quarkus, is that all dependencies (everything found under target/lib) are cached in a different layer than the actual application making rebuilds extra fast and extra small (when it comes to pushing).
+The extension `quarkus-container-image-jib` is powered by https://github.com/GoogleContainerTools/jib[Jib] for performing container image builds.
+The major benefit of using Jib with Quarkus, is that all dependencies (everything found under target/lib) are cached in a different layer than the actual application making rebuilds really fast and small (when it comes to pushing).
```
Comment (gsmet) [inline docs/src/main/asciidoc/container-image.adoc]:
```suggestion
The major benefit of using Jib with Quarkus is that all dependencies (everything found under target/lib) are cached in a different layer than the actual application making rebuilds really fast and small (when it comes to pushing).
```

File: docs/src/main/asciidoc/container-image.adoc [hunk 6]
```diff
@@ -69,69 +72,46 @@ The creation of such objects is being taken care of by the Quarkus kubernetes ex
...
-If the execution is set to `push` then after the build, the image will also be pushed to an image registry.
-The registry can be specified using `quarkus.container-image.registry` and will default to `docker.io`.
+To build a container image for your project, `quarkus.container-image.push=true` needs to be set using any of the ways that Quarkus supports.
+
+[source, subs=attributes+]
+----
+mvn clean package -Dquarkus.container-image.push=true
+----
 
+NOTE: If no registry is set (using `quarkus.container-image.registry`) then `docker.io` will be used by as the default.
```
Comment (gsmet) [inline docs/src/main/asciidoc/container-image.adoc]:
```suggestion
NOTE: If no registry is set (using `quarkus.container-image.registry`) then `docker.io` will be used as the default.
```

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java (hunk 4 above)
Comment (gsmet) [reply extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java]: I dunno if it's worth having an info for that.

File: docs/src/main/asciidoc/container-image.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/container-image.adoc]: If we do, please use ids for the links.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a couple of suggestions.

**Comment by geoand** [comment]
Thanks for going over the change. I'll probably take a look tomorrow or Monday morning

File: extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java (hunk 3 above)
Comment (geoand) [reply extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java]: I don't think this is an improvement to be honest

File: extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java (hunk 4 above)
Comment (geoand) [reply extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java]: I don't agree with the proposed change, but I do think that a log message should be used. After all the registry is `Optional` so when it's not set the user should know that some default was used

File: docs/src/main/asciidoc/container-image.adoc (hunk 1 above)
Comment (geoand) [reply docs/src/main/asciidoc/container-image.adoc]: Is my change the correct way to do it?

Comment (gsmet) [reply docs/src/main/asciidoc/container-image.adoc]: Not sure. Did you define ids on the paragraphs? I don’t see them in your diff but maybe they were preexisting (I’m on my phone).

Comment (geoand) [reply docs/src/main/asciidoc/container-image.adoc]: I didn't (I didn't even know you could). I'll update the PR in the morning

Comment (geoand) [reply docs/src/main/asciidoc/container-image.adoc]: Should be fixed with my latest change (I assume)

**Review by gsmet — APPROVED** [review]

**Pull request merged**